### PR TITLE
Issue 3440 is password auto complete disabled

### LINKED ISSUE #3440: password-mixin.js->isPasswordAutoCompleteDisabled - no longer needed.
As outlined in #2373, Firefox no longer respects `autocomplete=off`. [password-mixin.js-.isPasswordAutoCompleteDisabled](https://github.com/mozilla/fxa-content-server/blob/8c7a52148738f6a824b45532ff5f93627eb86d00/app/scripts/views/mixins/password-mixin.js#L26:L31) returns `true` if the relier is Sync. Since `autocomplete=off` is no longer respected, we might as well remove this function, any references to it, and any references to `autocomplete=off` on password fields.

Ref https://bugzilla.mozilla.org/show_bug.cgi?id=1173688

cc @rfk, @ckarlof.

### PR DESCRIPTION
**Pull request by feelcoder**
Issues #3440 commented

fixes #3440

### CODE REVIEW

File: app/scripts/templates/complete_reset_password.mustache [hunk 1]
```diff
@@ -46,7 +46,7 @@
 
       <div class="input-row password-row">
         <label class="label-helper"></label>
-        <input type="password" class="password" id="password" placeholder="{{#t}}New password{{/t}}" pattern=".{8,}" autofocus required {{#isPasswordAutoCompleteDisabled}}autocomplete="off"{{/isPasswordAutoCompleteDisabled}} />
```
Comment (shane-tomlinson) [inline app/scripts/templates/complete_reset_password.mustache]: Can you re-add the `required` attribute?

Comment (shane-tomlinson) [inline app/scripts/templates/complete_reset_password.mustache]: The `required` comment applies to all elements where it was removed.

File: app/scripts/views/complete_reset_password.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ define(function (require, exports, module) {
         isLinkDamaged: ! doesLinkValidate,
         isLinkExpired: isLinkExpired,
         isLinkValid: doesLinkValidate && ! isLinkExpired,
-        isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled()
+      //  isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled()
```
Comment (shane-tomlinson) [inline app/scripts/views/complete_reset_password.js]: This line should be complete removed from the source code. Remove the comma on the previous line as well.

File: app/scripts/views/force_auth.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ define(function (require, exports, module) {
       return {
         email: email,
         fatalError: getFatalErrorMessage(this, fatalError),
-        isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
+       // isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
```
Comment (shane-tomlinson) [inline app/scripts/views/force_auth.js]: Similar here, this line should be removed from the source code, with the comma on the previous line removed.

File: app/scripts/views/mixins/password-mixin.js [hunk 4]
```diff
@@ -23,12 +23,15 @@ define(function (require, exports, module) {
       this.focus(controlsSelector);
     },
 
+  // Not needed anymore
```
Comment (shane-tomlinson) [inline app/scripts/views/mixins/password-mixin.js]: It's considered bad practice to leave code that is no longer needed in the tree. Can you remove this function from the source code? We can always go back through git history if we need the source again in the future.

File: app/scripts/views/mixins/password-mixin.js [hunk 5]
```diff
@@ -39,17 +42,12 @@ define(function (require, exports, module) {
...
-            .attr('autocorrect', 'off').attr('autocapitalize', 'off');
+          passwordField.attr('type', 'text').attr('autocorrect', 'off').attr('autocapitalize', 'off');
           this.logViewEvent('password.visible');
         } else {
           passwordField.attr('type', 'password');
-          if (this.isPasswordAutoCompleteDisabled()) {
-            passwordField.attr('autocomplete', 'off');
-          } else {
-            passwordField.removeAttr('autocomplete')
+          passwordField.removeAttr('autocomplete')
```
Comment (shane-tomlinson) [inline app/scripts/views/mixins/password-mixin.js]: It looks like this line can probably be combined with line 49

File: app/scripts/views/settings/change_password.js [hunk 6]
```diff
@@ -24,11 +24,12 @@ define(function (require, exports, module) {
     className: 'change-password',
     viewName: 'settings.change-password',
 
-    context: function () {
+  /*  context: function () {
```
Comment (shane-tomlinson) [inline app/scripts/views/settings/change_password.js]: Kill the commented out code with fire.

File: app/scripts/views/sign_in.js [hunk 7]
```diff
@@ -66,7 +66,7 @@ define(function (require, exports, module) {
         chooserAskForPassword: this._suggestedAccountAskPassword(suggestedAccount),
         email: email,
         error: this.error,
-        isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
+       // isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
```
Comment (shane-tomlinson) [inline app/scripts/views/sign_in.js]: Remove the commented out code here too.

File: app/scripts/views/sign_up.js [hunk 8]
```diff
@@ -162,7 +162,7 @@ define(function (require, exports, module) {
         isAmoMigration: this.isAmoMigration(),
         isCustomizeSyncChecked: relier.isCustomizeSyncChecked(),
         isEmailOptInVisible: this._isEmailOptInEnabled(),
-        isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
+       // isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
```
Comment (shane-tomlinson) [inline app/scripts/views/sign_up.js]: I'll be you know what I'm going to say here.

**Comment by shane-tomlinson** [comment]
@feelcoder - this is a fantastic start! You'll also have to update the unit tests, at a minimum [app/tests/spec/views/mixins/password-mixin.js](https://github.com/mozilla/fxa-content-server/blob/master/app/tests/spec/views/mixins/password-mixin.js) needs to be updated. All tests pertaining to `autocomplete` can be removed, it looks like we failed to add a test for `isPasswordAutoCompleteDisabled`.

Once you make the updates, can you squash the commits into a single commit, and update the description to something like:

```
fix(client): Remove password `autocomplete=off` functionality
```

File: app/scripts/views/force_auth.js (hunk 3 above)
Comment (feelcoder) [inline app/scripts/views/force_auth.js]: @shane-tomlinson  If I remove the comma here, it will produce an error, since its is not the end of the return

Comment (shane-tomlinson) [inline app/scripts/views/force_auth.js]: Ah yes, you are right.

File: app/scripts/templates/sign_up.mustache [hunk 9]
```diff
@@ -33,12 +33,12 @@
...
 
       <div class="input-row password-row">
-        <input id="password" type="password" class="password tooltip-below" placeholder="{{#t}}Password{{/t}}" value="{{ password }}" pattern=".{8,}" required {{#isPasswordAutoCompleteDisabled}}autocomplete="off"{{/isPasswordAutoCompleteDisabled}} {{#shouldFocusPassword}}autofocus{{/shouldFocusPassword}} />
+        <input id="password" type="password" class="password tooltip-below" placeholder="{{#t}}Password{{/t}}" value="{{ password }}" pattern=".{8,}" required {{#shouldFocusPassword}}autofocus{{/shouldFocusPassword}} />
         <input id="show-password" type="checkbox" class="show-password" aria-controls="password" tabindex="-1" data-novalue />
         <label for="show-password" class="show-password-label">
           {{#t}}Show{{/t}}
         </label>
-        <div class="input-help input-help-focused input-help-signup">{{#t}}Must be at least 8 characters{{/t}}</div>
+         <div class="input-help input-help-focused input-help-signup">{{#t}}Must be at least 8 characters{{/t}}</div>
```
Comment (vladikoff) [inline app/scripts/templates/sign_up.mustache]: extra space needs to be removed

File: app/scripts/templates/complete_reset_password.mustache [hunk 10]
```diff
@@ -46,7 +46,7 @@
...
+        <input type="password" class="password" id="password" placeholder="{{#t}}New password{{/t}}" pattern=".{8,}" required autofocus/>
```
Comment (vladikoff) [inline app/scripts/templates/complete_reset_password.mustache]: add a space here after `autofocus` to be consistent with other tags below. 

Should be `autofocus />`

File: app/scripts/views/mixins/password-mixin.js [hunk 11]
```diff
@@ -39,17 +32,11 @@ define(function (require, exports, module) {
...
-            .attr('autocorrect', 'off').attr('autocapitalize', 'off');
+          passwordField.attr('type', 'text').attr('autocorrect', 'off').attr('autocapitalize', 'off');
           this.logViewEvent('password.visible');
         } else {
-          passwordField.attr('type', 'password');
-          if (this.isPasswordAutoCompleteDisabled()) {
-            passwordField.attr('autocomplete', 'off');
-          } else {
-            passwordField.removeAttr('autocomplete')
+          passwordField.attr('type', 'password').removeAttr('autocomplete')
```
Comment (vladikoff) [inline app/scripts/views/mixins/password-mixin.js]: Do you still need `.removeAttr('autocomplete')` as part of this?

**Comment by vladikoff** [comment]
@feelcoder Awesome work! This is almost ready! I left 3 comments above to fix.

As Shane mentioned above : 

> Once you make the updates, can you squash the commits into a single commit, and update the description to something like:

```
fix(client): Remove password `autocomplete=off` functionality

Fixes #3440 
```

The 3 commits we see above should become one "squashed commit". This guide will help you with that: https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase-i

File: app/scripts/views/mixins/password-mixin.js [hunk 12]
```diff
@@ -23,13 +23,6 @@ define(function (require, exports, module) {
       this.focus(controlsSelector);
     },
 
-    isPasswordAutoCompleteDisabled: function () {
-      // Sync users should never be allowed to save their password. If they
-      // were, it would end in this weird situation where sync users ask to
-      // save their sync password to sync before sync is setup.
-      return this.relier.isSync();
-    },
```
Comment (vladikoff) [inline app/scripts/views/mixins/password-mixin.js]: :+1:

File: app/scripts/views/mixins/password-mixin.js (hunk 11 above)
Comment (vladikoff) [inline app/scripts/views/mixins/password-mixin.js]: Looks like you still need this functionality here for `setPasswordVisibilityFromButton`. 

We want to set `autocomplete` when the user presses the "Show" button. This means you need to revert the changes you made here in the `setPasswordVisibilityFromButton` function.

File: app/tests/spec/views/mixins/password-mixin.js [hunk 13]
```diff
@@ -63,37 +61,18 @@ define(function (require, exports, module) {
...
-
-        sinon.stub(relier, 'isSync', function () {
-          return true;
-        });
-
-        view.$('#show-password').removeAttr('checked');
-        view.setPasswordVisibilityFromButton('#show-password');
-        assert.equal(view.$('#password').attr('type'), 'password');
-        assert.equal(view.$('#password').attr('autocomplete'), 'off');
-      });
```
Comment (vladikoff) [inline app/tests/spec/views/mixins/password-mixin.js]: this is good!

File: app/tests/spec/views/mixins/password-mixin.js [hunk 14]
```diff
@@ -63,37 +61,18 @@ define(function (require, exports, module) {
...
     });
 
     describe('clicking on the show button', function () {
```
Comment (vladikoff) [inline app/tests/spec/views/mixins/password-mixin.js]: undo the changes below in this test , we still need this

**Comment by feelcoder** [comment]
Yes, @vladikoff, I will do that. I thought i have first squashed it. But it does not seem. I will check again how to squash commits

File: app/scripts/views/mixins/password-mixin.js (hunk 11 above)
Comment (feelcoder) [inline app/scripts/views/mixins/password-mixin.js]: @vladikoff I did not modify the function setPasswordVisibbilityFromButton. Do you mean rather in the setPasswordVisibility?

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: ce9e9a1680d0424e8468875b2fa7afdbe86ae6b1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

File: app/scripts/templates/complete_reset_password.mustache [hunk 15]
```diff
@@ -46,7 +46,7 @@
 
       <div class="input-row password-row">
         <label class="label-helper"></label>
-        <input type="password" class="password" id="password" placeholder="{{#t}}New password{{/t}}" pattern=".{8,}" autofocus required {{#isPasswordAutoCompleteDisabled}}autocomplete="off"{{/isPasswordAutoCompleteDisabled}} />
+        <input type="password" class="password" id="password" placeholder="{{#t}}New password{{/t}}" pattern=".{8,}" required autofocus />
```
Comment (vladikoff) [inline app/scripts/templates/complete_reset_password.mustache]: Looks like you got a trailing space here now after `/>`. Please remove it

File: app/scripts/views/mixins/password-mixin.js [hunk 16]
```diff
@@ -39,17 +32,10 @@ define(function (require, exports, module) {
       try {
         var passwordField = this.$('.password');
         if (isVisible) {
-          passwordField.attr('type', 'text').attr('autocomplete', 'off')
-            .attr('autocorrect', 'off').attr('autocapitalize', 'off');
```
Comment (vladikoff) [inline app/scripts/views/mixins/password-mixin.js]: We still need this and the changes below. No need to make any changes to `setPasswordVisibilityFromButton`

File: app/scripts/templates/complete_reset_password.mustache (hunk 15 above)
Comment (vladikoff) [inline app/scripts/templates/complete_reset_password.mustache]: Comment still applies ^

File: app/scripts/views/force_auth.js [hunk 17]
```diff
@@ -130,9 +130,14 @@ define(function (require, exports, module) {
...
       return {
+<<<<<<< HEAD
+        email: email,
+        fatalError: getFatalErrorMessage(this, fatalError),
+=======
         email: this.relier.get('email'),
         fatalError: getFatalErrorMessage(this, this._fatalError),
         isPasswordAutoCompleteDisabled: this.isPasswordAutoCompleteDisabled(),
+>>>>>>> c98f30d67969f07e229f28a1eb73af019e6be41a
         password: this._formPrefill.get('password')
```
Comment (vladikoff) [inline app/scripts/views/force_auth.js]: ^ Above needs to be fixed, you need to resolve the conflict here. Basically change this file to have:
```
    context: function () {
      return {
        email: this.relier.get('email'),
        fatalError: getFatalErrorMessage(this, this._fatalError),
        password: this._formPrefill.get('password')
      };
    },
```

File: app/scripts/views/mixins/password-mixin.js (hunk 16 above)
Comment (vladikoff) [inline app/scripts/views/mixins/password-mixin.js]: Comment still applies ^

**Comment by vladikoff** [comment]
@feelcoder Great work getting this rebased! now we just need to fix up the issues i mentioned above.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

File: app/scripts/views/mixins/password-mixin.js (hunk 16 above)
Comment (feelcoder) [inline app/scripts/views/mixins/password-mixin.js]: @vladikoff adding back all the changes below will include the deleted function isPasswordAutoCompleteDisabled()
'else {
          passwordField.attr('type', 'password');
          if (this.isPasswordAutoCompleteDisabled()) {
            passwordField.attr('autocomplete', 'off');
          } else { '

Comment (vladikoff) [inline app/scripts/views/mixins/password-mixin.js]: Ah I see, so this should be something like this:

```js
        if (isVisible) {
          passwordField.attr('type', 'text').attr('autocomplete', 'off')
            .attr('autocorrect', 'off').attr('autocapitalize', 'off');
          this.logViewEvent('password.visible');
        } else {
          passwordField.attr('type', 'password');
          passwordField.removeAttr('autocomplete')		
               .removeAttr('autocorrect').removeAttr('autocapitalize');
          this.logViewEvent('password.hidden');
        }
```

Your new change is missing `.removeAttr('autocomplete')` as part of the function call.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

File: app/scripts/views/mixins/password-mixin.js [hunk 18]
```diff
@@ -44,12 +37,8 @@ define(function (require, exports, module) {
...
         } else {
           passwordField.attr('type', 'password');
-          if (this.isPasswordAutoCompleteDisabled()) {
-            passwordField.attr('autocomplete', 'off');
-          } else {
-            passwordField.removeAttr('autocomplete')
-              .removeAttr('autocorrect').removeAttr('autocapitalize');
-          }
+          passwordField.removeAttr('autocomplete')      
+               .removeAttr('autocorrect').removeAttr('autocapitalize');
```
Comment (pdehaan) [inline app/scripts/views/mixins/password-mixin.js]: I think you can probably reduce this to a single line and just do something like this:

```js
passwordField.removeAttr('autocomplete autocorrect autocapitalize');
```

---

> #### [`.removeAttr( attributeName )`](https://api.jquery.com/removeAttr/#removeAttr-attributeName)
> _An attribute to remove; as of version 1.7, it can be a space-separated list of attributes._

Comment (vladikoff) [inline app/scripts/views/mixins/password-mixin.js]: @feelcoder up to you if you want to make this change or leave it like it is right now.

File: app/tests/spec/views/mixins/password-mixin.js [hunk 19]
```diff
@@ -12,7 +12,6 @@ define(function (require, exports, module) {
   var Metrics = require('lib/metrics');
   var PasswordMixin = require('views/mixins/password-mixin');
   var Relier = require('models/reliers/relier');
-  var sinon = require('sinon');
```
Comment (vladikoff) [inline app/tests/spec/views/mixins/password-mixin.js]: good catch !

File: app/tests/spec/views/mixins/password-mixin.js [hunk 20]
```diff
@@ -54,7 +53,6 @@ define(function (require, exports, module) {
         view.$('#show-password').attr('checked', 'checked');
         view.setPasswordVisibilityFromButton('#show-password');
         assert.equal(view.$('#password').attr('type'), 'text');
-        assert.equal(view.$('#password').attr('autocomplete'), 'off');
```
Comment (vladikoff) [inline app/tests/spec/views/mixins/password-mixin.js]: @feelcoder do tests fail if you have this included?

**Comment by vladikoff** [comment]
@feelcoder Here are some errors reported by our Travis CI checker:

```
/home/travis/build/mozilla/fxa-content-server/app/scripts/views/mixins/password-mixin.js
  40:51  error  Trailing spaces not allowed  no-trailing-spaces
```

```
/home/travis/build/mozilla/fxa-content-server/app/tests/spec/views/mixins/password-mixin.js
   69:4  error  Expected indentation of 4 space characters but found 3  indent
   70:7  error  Expected indentation of 5 space characters but found 6  indent
   82:7  error  Expected indentation of 5 space characters but found 6  indent
   95:7  error  Expected indentation of 5 space characters but found 6  indent
  107:5  error  Expected indentation of 3 space characters but found 4  indent
✖ 6 problems (6 errors, 0 warnings)
```

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

File: app/scripts/views/mixins/password-mixin.js (hunk 18 above)
Comment (feelcoder) [inline app/scripts/views/mixins/password-mixin.js]: @pdehaan, I did what you said and when I ran grunt lint it gave me Warning: Task "jscs:app" failed. So, I had to undo and it came out without errors. Beginning to understand the importance of styling @vladikoff

File: app/scripts/templates/sign_up.mustache [hunk 21]
```diff
@@ -39,13 +39,12 @@
...
       <div class="input-row password-row">
-        <input id="password" type="password" class="password tooltip-below" placeholder="{{#t}}Password{{/t}}" value="{{ password }}" pattern=".{8,}" required {{#isPasswordAutoCompleteDisabled}}autocomplete="off"{{/isPasswordAutoCompleteDisabled}} {{#shouldFocusPassword}}autofocus{{/shouldFocusPassword}} />
+        <input id="password" type="password" class="password tooltip-below" placeholder="{{#t}}Password{{/t}}" value="{{ password }}" pattern=".{8,}" required {{#shouldFocusPassword}}autofocus{{/shouldFocusPassword}} />
         <input id="show-password" type="checkbox" class="show-password" aria-controls="password" tabindex="-1" data-novalue />
         <label for="show-password" class="show-password-label">
           {{#t}}Show{{/t}}
         </label>
-        <div class="input-help input-help-focused input-help-signup">{{#t}}Must be at least 8 characters{{/t}}</div>
-
+         <div class="input-help input-help-focused input-help-signup">{{#t}}Must be at least 8 characters{{/t}}</div>
```
Comment (vladikoff) [inline app/scripts/templates/sign_up.mustache]: We don't need an extra space before this `<div` tag.

**Comment by vladikoff** [comment]
@feelcoder looks like this is almost ready! 

1. Fix the small issue in https://github.com/mozilla/fxa-content-server/pull/3597/files#r57460225
2. Squash the commits into one. Here is a guide: http://push.cwcon.org/learn/squashing-commits.html 
Fix the commit message to say `fix(client): isPasswordAutoCompleteDisabled removed`

Once that is complete we should have 1 commit instead of 9 and we will merge your pull request!

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 37f2d06c3b3550a7a0d7bc5a8751f70922acdb87
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by vladikoff** [comment]
Hey, so this is getting close in terms of the commit messages.

You probably need to rebase your branch against `origin/master`. 

So when you are on your branch do : `git rebase origin/master` that will pick up the latest changes.

Here's also another guide to squashing multiple commits: https://youtu.be/qh9KtjfjzCU?t=150

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: ffa14ce05a39f8150348fd4b6df19b43cc68f4e7
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by vladikoff** [comment]
@feelcoder Looks perfect ! Waiting for CI tests to pass :+1:

**Comment by pdehaan** [comment]
@feelcoder great work, congrats!
